posthog/tasks/tasks.py

```diff
@@ -930,6 +930,187 @@ def background_delete_model_task(
+        )
+
+        # Query ClickHouse for flag usage since last sync
+        # Limit for insurance against large datasets and memory issues during a surge
+        result = sync_execute(
+            """
+            SELECT
+                team_id,
+                JSONExtractString(properties, '$feature_flag') as flag_key,
+                max(timestamp) as last_called_at,
+                count() as call_count
+            FROM events
+            PREWHERE event = '$feature_flag_called'
+            WHERE timestamp > %(last_sync_timestamp)s
+              AND timestamp <= %(current_sync_timestamp)s
+              AND JSONExtractString(properties, '$feature_flag') != ''
+            GROUP BY team_id, flag_key
+            ORDER BY last_called_at DESC
+            LIMIT %(limit)s
+            """,
+            {
+                "last_sync_timestamp": last_sync_timestamp,
+                "current_sync_timestamp": current_sync_timestamp,
+                "limit": settings.FEATURE_FLAG_LAST_CALLED_AT_SYNC_CLICKHOUSE_LIMIT,
+            },
+        )
+
+        if not result:
+            # Update checkpoint even if no results
+            redis_client.set(FEATURE_FLAG_LAST_CALLED_SYNC_KEY, current_sync_timestamp.isoformat())
+            logger.info(
+                "Feature flag sync completed with no events",
+                duration_seconds=(timezone.now() - start_time).total_seconds(),
+            )
+            return
+
+        # Collect flags for bulk update
+        flags_to_update = []
+
+        # Get latest timestamp for checkpoint, fallback to current if all None
+        checkpoint_timestamp = max((row[2] for row in result if row[2]), default=current_sync_timestamp)
+
+        # Build lookup map of (team_id, key) -> timestamp from ClickHouse results
+        flag_updates = {(row[0], row[1]): row[2] for row in result}
+
+        # Batch fetch all relevant flags in a single query
+        team_ids = list({row[0] for row in result})
+        flag_keys = list({row[1] for row in result})
+
+        flags = FeatureFlag.objects.filter(team_id__in=team_ids, key__in=flag_keys)
+
+        for flag in flags:
+            new_timestamp = flag_updates.get((flag.team_id, flag.key))
+            if new_timestamp and (flag.last_called_at is None or flag.last_called_at < new_timestamp):
+                flag.last_called_at = new_timestamp
+                flags_to_update.append(flag)
+
+        # Perform bulk update
+        updated_count = 0
+        if flags_to_update:
+            try:
+                FeatureFlag.objects.bulk_update(
+                    flags_to_update,
+                    ["last_called_at"],
+                    batch_size=settings.FEATURE_FLAG_LAST_CALLED_AT_SYNC_BATCH_SIZE,
+                )
+                updated_count = len(flags_to_update)
+            except Exception as e:
+                capture_exception(e, additional_properties={"flags_count": len(flags_to_update)})
```

Comment:
I'd recommend tagging the exception in case you ever need to look it up in PostHog. E.g.:
```suggestion
                capture_exception(e, additional_properties={"tag": "sync_feature_flag_last_called", "flags_count": len(flags_to_update)})
```